core/src/components/datetime/datetime.tsx

```diff
@@ -841,17 +841,35 @@ export class Datetime implements ComponentInterface {
         const root = this.el!.shadowRoot!;
 
         /**
-         * Get the element that is in the center of the calendar body.
-         * This will be an element inside of the active month.
+         * Get the elements that are in the center of the calendar body.
+         * The topmost of these will be elements inside of the active month.
+         *
+         * We filter out popovers to exclude the time picker, if present,
+         * since it sits on top of the entire screen. This handles the
+         * time picker being open when the datetime's value is changed
+         * programmatically, for example.
+         */
+        const elementsAtCenter = root
+          .elementsFromPoint(box.x + box.width / 2, box.y + box.height / 2)
+          .filter((el) => el.tagName !== 'ION-POPOVER');
```

Comment:
This is technically a bug in `main`, right?

Reply:
Sort of, but it's very slight. The issue arises when you try to scroll months with the time picker open. In main, this is only possible by manually calling `.click()` on the next/prev month buttons, which are in the shadow DOM, so you'd have to really be going out of your way.

Reply:
Makes sense! The bug seems more noticeable with this change, so probably good to keep the fixes together.